vars/piperStageWrapper.groovy

```diff
@@ -0,0 +1,107 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.ConfigurationLoader
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'piperStageWrapper'
+
+void call(Map parameters = [:], body) {
+
+    def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+    def utils = parameters.juStabUtils ?: new Utils()
+
+    def stageName = parameters.stageName?:env.STAGE_NAME
+
+    // load default & individual configuration
+    Map config = ConfigurationHelper.newInstance(this)
+        .loadStepDefaults()
+        .mixin(ConfigurationLoader.defaultStageConfiguration(this, stageName))
+        .mixinGeneralConfig(script.commonPipelineEnvironment)
+        .mixinStageConfig(script.commonPipelineEnvironment, stageName)
+        .mixin(parameters)
+        .use()
+
+    stageLocking(config) {
+        withNode(config) {
+            try {
+                utils.unstashAll(config.stashContent)
+
+                if (Boolean.valueOf(env.ON_K8S) && containerMap.size() > 0) {
+                    withEnv(["POD_NAME=${stageName}"]) {
+                        dockerExecute(script: script, containerMap: containerMap) {
+                            executeStage(body, stageName, config)
+                        }
+                    }
+                } else {
+                    executeStage(body, stageName, config)
+                }
+            } finally {
+                deleteDir()
+            }
+        }
+    }
+}
+
+String stageExitFilePath(String stageName, Map config) {
+    return "${config.extensionLocation}${stageName.replace(' ', '_').toLowerCase()}.groovy"
```

Comment:
We call our stages like buildFrontend and expect buildFrontend.groovy as file. However, with the conversion toLowerCase it is now buildfrontend.groovy

Reply:
I guess we should then ignore the case in the comparison.
@daniel-kurzynski is replacing spaces okay from your end?

Reply:
no space replacements anymore. Stage is used as is ...